src/protocol/securemul.rs

```diff
@@ -69,15 +69,18 @@ impl<'a, S: Step, N: Network<S>> SecureMul<'a, S, N> {
         // notify helper on the right that we've computed our value
         channel
             .send(
-                channel.identity().peer(Direction::Right),
+                self.gateway.identity().peer(Direction::Right),
```

Comment:
Honestly, I think the better solution here is to just change the internal state that SecureMul keeps around. Nothing else keeps a special pointer to the gateway and the prss, everything else just gets what it needs directly from the ProtocolContext. I'd recommend updating SecureMul to do the same.

Reply:
I am leaning towards the same conclusion, but want to postpone that change until I talk to Martin about it - we had some considerations for this struct that led to this API. For now, I removed `identity` method from `Gateway` and just keep the role inside `SecureMul`